### PR TITLE
feat: kernel: use specified proof type when verifying PoSts

### PR DESCRIPTION
**Pull request by arajasek**
We no longer have only one PoSt proof type corresponding to each Seal proof type. As a result, when verifying PoSts, we must supply the proof type that the proof "announces" as the public Replica's PoSt proof type. This is still secure as proofs code checks that all replicas have the same proof type, and actors code confirms that the proof type of the first PoSt is valid.

If we wanted to, we could revert this change in FVM4, at which point we'll have a one PoSt per Seal restriction again. If we _really_ wanted to, we could make this code network version dependent, and only apply this change to nv19, but I think that belongs in actors code.

### CODE REVIEW

**Review by Stebalien — APPROVED** [review]

File: fvm/src/kernel/default.rs [hunk 1]
```diff
@@ -1173,8 +1162,10 @@ fn verify_post(verify_info: &WindowPoStVerifyInfo) -> Result<bool> {
     // Necessary to be valid bls12 381 element.
     randomness[31] &= 0x3f;
 
+    let proof_type = proofs[0].post_proof;
```
Comment (Stebalien) [inline fvm/src/kernel/default.rs:1165]: I'd slightly prefer to validate that all of the proof types here are the same.

File: fvm/src/kernel/default.rs [hunk 2]
```diff
@@ -1114,18 +1107,14 @@ fn get_required_padding(
...
 ) -> Result<BTreeMap<SectorId, PublicReplicaInfo>> {
     let replicas = src
         .iter()
         .map::<core::result::Result<(SectorId, PublicReplicaInfo), String>, _>(
             |sector_info: &SectorInfo| {
                 let commr = commcid::cid_to_replica_commitment_v1(&sector_info.sealed_cid)?;
-                let proof = match typ {
-                    ProofType::Winning => sector_info.proof.registered_winning_post_proof()?,
-                    ProofType::Window => sector_info.proof.registered_window_post_proof()?,
-                };
```
Comment (Stebalien) [inline fvm/src/kernel/default.rs:1127]: Is there some way to quickly check that the proof type is valid for the given sectors?

**Review by Stebalien — APPROVED** [review]

**Comment by Stebalien** [comment]
My concern is that... that code is _very_ far away from this code. I'd rather just have the extra checks, honestly.

**Comment by arajasek** [comment]
@Stebalien Yup, happy to do so.

**Review by Stebalien — APPROVED** [review]
Ah, I see why you weren't thrilled about this. But I think it's safer to have for now...

**Pull request merged**